### PR TITLE
Hub: Fixes trailing spaces in api

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: hub/api/pkg/sync/sync.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func (s *SyncService) updateResources(job model.SyncJob, repo git.Repo, res []gi
 			for _, t := range v.Tags {
 				tag := model.Tag{Name: t, CategoryID: others.ID}
```
Comment (SM43) [inline hub/api/pkg/sync/sync.go:171]: here also spaces needs to be trimmed

Comment (SM43) [reply hub/api/pkg/sync/sync.go:171]: `tag` is getting inserted so it should be trimmed

File: hub/api/pkg/sync/sync.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func (s *SyncService) updateResources(job model.SyncJob, repo git.Repo, res []gi
...
 
-				txn.Model(&model.Tag{}).Where(&model.Tag{Name: t}).FirstOrCreate(&tag)
+				txn.Model(&model.Tag{}).Where(&model.Tag{Name: strings.TrimSpace(t)}).FirstOrCreate(&tag)
```
Comment (sthaha) [inline hub/api/pkg/sync/sync.go:173]: this doesn't still address the actual cause, does it? the v.Tags contain spaces. I think the Trim must be applied to the parsing logic.

**Pull request merged**